### ISSUE TITLE
4.9 build fails with --enable-shared=no --enable-static=yes

### ISSUE THREAD

**Issue by Polynomial-C** [shadow-maint/shadow#387]
```
libtool: link: x86_64-pc-linux-gnu-gcc -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -o sulogin sulogin.o  -Wl,--as-needed ../libmisc/.libs/libmisc.a ../lib/.libs/libshadow.a -lcrypt
x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I..  -I../lib -I../libmisc -I../libsubid   -march=native -mtune=native -O2 -pipe -c -o list_subid_ranges-list_subid_ranges.o `test -f 'list_subid_ranges.c' || echo './'`list_subid_ranges.c
/bin/sh ../libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc  -march=native -mtune=native -O2 -pipe  -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--as-needed -o list_subid_ranges list_subid_ranges-list_subid_ranges.o ../lib/libshadow.la ../libmisc/libmisc.la ../libsubid/libsubid.la        -lcrypt  -ldl
libtool:   error: cannot find name of link library for '../libsubid/libsubid.la'
make[2]: *** [Makefile:1085: list_subid_ranges] Error 1
make[2]: Leaving directory '/var/tmp/portage/sys-apps/shadow-4.9/work/shadow-4.9/src'
make[1]: *** [Makefile:452: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/sys-apps/shadow-4.9/work/shadow-4.9'
make: *** [Makefile:380: all] Error 2
```

**Comment by hallyn** [comment]
Thanks - looks like a duplicate of #386 so I'll close this one.

**Comment by Polynomial-C** [comment]
I saw issue #386 before I filed this one. But the errors differ so much that I decided this is not the same issue. My issue is that the libtool archive file cannot be found (although it's present in the `libsubid` subdir) while #386 is about a header file that cannot be found.

For the record, I have this issue on Gentoo Linux.

**Comment by hallyn** [comment]
This does seem to have been a dup of #386, and in my test on a gentoo box is fixed by the same commit.

Are you building a new ebuild?  Would it be easier if I create a 4.9.1 release, or is it just as easy for you to apply a patch as part of the ebuild?

**Comment by Polynomial-C** [comment]
No, this is not fixed with commit f4a84efb468b8be21be124700ce35159c444e9d6 and commit 537b8cd90be7b47b45c45cfd27765ef85eb0ebf1 being applied. Build breaks with the same error message I have posted in my initial comment.

But I have dug a bit deeper now and found our current Gentoo ebuild using the following configure options:

`--enable-shared=no --enable-static=yes`

After removing these two switches, shadow-4.9 now compiles without a problem. I still suggest to investigate why these two options make the build fail now.

**Comment by hallyn** [comment]
Thanks!
